airflow/utils/log/file_task_handler.py

```diff
@@ -62,19 +62,29 @@ def __init__(self, base_log_folder: str, filename_template: str | None = None):
                 # handler, not the one that calls super()__init__.
                 stacklevel=(2 if type(self) == FileTaskHandler else 3),
             )
+        self.maintain_propagate: bool = False
 
     def set_context(self, ti: TaskInstance) -> None | SetContextPropagate:
         """
         Provide task_instance context to airflow task handler.
 
+        Generally speaking returns None.  But if attr `maintain_propagate` has
+        been set to propagate, then returns sentinel MAINTAIN_PROPAGATE. This
+        has the effect of overriding the default behavior to set `propagate`
+        to False whenever set_context is called.  At time of writing, this
+        functionality is only used in unit testing.
+
         :param ti: task instance object
         """
-        local_loc = self._init_file(ti)
-        self.handler = NonCachingFileHandler(local_loc, encoding="utf-8")
-        if self.formatter:
-            self.handler.setFormatter(self.formatter)
-        self.handler.setLevel(self.level)
-        return None
+        from airflow.models.taskinstance import TaskInstance
+
+        if isinstance(ti, TaskInstance):
+            local_loc = self._init_file(ti)
+            self.handler = NonCachingFileHandler(local_loc, encoding="utf-8")
+            if self.formatter:
+                self.handler.setFormatter(self.formatter)
+            self.handler.setLevel(self.level)
```

Comment:
When is this not true…?

Reply:
i think it came up when i was mocking a TI.... and  it doesn't hurt anybody.... but i've removed it for now let's see what the tests say

Reply:
It does hurt somebody if they end up scratching their head to make sense of it!

Reply:
well... i guess there are degrees of hurt...

anyway, it was needed because something in existing test was mocking a TI and the actual methods don't like the mock obj

and since i restored normal logging config, now the mocked ti is hitting handler. anyway, mocked this method away so non issue now.  removed the if, and tests should go green again.